nvidia.go

```diff
@@ -0,0 +1,116 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const ldconfPath = "/etc/ld.so.conf"
+
+func setupNvidiaDriver(containerRootfs string) (err error) {
+	bins, libs, err := getNvidiaFiles()
+	if err != nil {
+		return
+	}
+
+	// mount driver binaries
+	for _, binFile := range bins {
+		if err = mount(binFile, containerRootfs+binFile, "bind", unix.MS_BIND|unix.MS_REC, "rbind"); err != nil {
+			return
+		}
+	}
+
+	// mount driver libraries
+	libDirsMap := make(map[string]interface{})
```

Comment:
are you using a map to skip duplicates? if so, you should check if the path already exist before bind-mount it again (next lines)

Reply:
I use a map to skip duplicate lib dirs but not lib files. These files are not duplicated. The mount function also checks if the path already exists.
"nvidia-container-cli list -l" output like follow:
```
/usr/lib64/libnvidia-ml.so.418.116.00
/usr/lib64/libnvidia-cfg.so.418.116.00
/usr/lib64/vdpau/libvdpau_nvidia.so.418.116.00
/usr/lib64/libnvidia-encode.so.418.116.00
...
```
mount every file
Dirs "/usr/lib64/" "/usr/lib64/vdpau/" are saved to use as args of ldconfig and write into "/etc/ld.so.conf"